pkg/bgpv1/test/links.go

```diff
@@ -0,0 +1,119 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package test
+
+import (
+	"net"
+	"net/netip"
+
+	"github.com/vishvananda/netlink"
+)
+
+// Note : this will work only in linux environment and requires CAP_NET_ADMIN privilege
+
+// dummyInterfaces contains IP addresses for the link
+type dummyInterfaces struct {
+	ipv4 netip.Prefix
+	ipv6 netip.Prefix
+}
+
+// default dummy links on which bgp sessions are communicating
+var (
+	ciliumLink    = "cilium-bgp"
+	instance1Link = "instance1"
+	instance2Link = "instance2"
+
+	dummies = map[string]dummyInterfaces{
+		// link used by BGP Control Plane
+		ciliumLink: {
+			ipv4: netip.MustParsePrefix("172.16.100.1/32"),
+			ipv6: netip.MustParsePrefix("a::1/128"),
+		},
+		// link used by gobgp instance 1
+		instance1Link: {
+			ipv4: netip.MustParsePrefix("172.16.100.2/32"),
+			ipv6: netip.MustParsePrefix("a::2/128"),
+		},
+		// link used by gobgp instance 2
+		instance2Link: {
+			ipv4: netip.MustParsePrefix("172.16.100.3/32"),
+			ipv6: netip.MustParsePrefix("a::3/128"),
```

Comment:
These IP addresses may overlap with someone's machine. That means this test may fail in such an environment. So, it's better to create dummy interfaces inside the dedicated network namespace. So that we can use whatever address we want there.

Reply:
This turned out to be more tricky, not exactly sure on how to fix it.

Setting netns from within go test does not work well, as new go routines are not part of same netns. 
https://github.com/containernetworking/plugins/blob/main/pkg/ns/README.md 

Links and addrs are created inside a new namespace, but gobgp starts listening in parent ns.

Reply:
I wonder, once privilege tests are moved from ginkgo to github actions, we can wrap this via shell in github workflow ?

```
ip netns add bgp-test
ip netns bgp-test exec go test <bgp tests>
ip netns delete bgp-test
```

Reply:
> Spin up multiple GoBGP instances locally with different port numbers. We first need to support changing the peer's port number to implement this.

Currently, that is happening. Peering gobgp instances are started with different port number. Since they are the ones connecting to BGP CP gobgp, it works fine. 

> Run peer GoBGP daemons separately in the different processes in different netns.

I'd like to have control over gobgp instances from within test driver - so we can inspect its internal state. I have a feeling this is going to be complicated if we start gobgp from outside the scope of test driver. 

> Make IP addresses configurable

I am not sure what you mean by this one ? Like pass IP in test environment  ?

Maybe short term, I can make IPs overridable via env settings. If that is not enough, then look at creating dedicated github workflow to create netns before running this test case.

Reply:
> Like pass IP in test environment ?

Yep, that's right. Seems like you already did this in the latest commit. We've also talked out-of-band and agreed to go with a configurable IP address in the short term.